internal/service/common/utils/repository.go

```diff
@@ -9,96 +9,212 @@ import (
 	// Build sql query
 	var record T
-	var records []T
 	if columns == nil {
-		tags := GetAllDBTagsFromStruct(record)
+		tags := GetAllDBTagsFromStruct(record, IncludeNilValues)
 		columns = tags.Columns()
 	}
 
-	query, args, err := psql.Select(
+	sql, args, err := psql.Select(
 		sm.Columns(columns...),
 		sm.From(record.TableName()),
+		sm.Where(expression),
 	).Build()
 	if err != nil {
 		return []T{}, fmt.Errorf("failed to build query: %w", err)
 	}
 
+	slog.Debug("executing statement", "sql", sql, "args", args)
+
 	// Run query
-	rows, _ := db.Query(ctx, query, args...) // note: err is passed on to Collect* func so we can ignore this
-	records, err = pgx.CollectRows(rows, pgx.RowToStructByNameLax[T])
+	rows, _ := db.Query(ctx, sql, args...) // note: err is passed on to Collect* func so we can ignore this
+	records, err := pgx.CollectRows(rows, pgx.RowToStructByNameLax[T])
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
+			slog.Debug("no records found", "table", record.TableName(), "expression", expression)
 			return []T{}, nil
 		}
-		return []T{}, fmt.Errorf("failed to call database: %w", err)
+		return []T{}, fmt.Errorf("failed to execute query: %w", err)
 	}
 
-	slog.Info("records found", "count", len(records), "table", record.TableName())
+	slog.Debug("records found", "table", record.TableName(), "expression", expression, "count", len(records))
 	return records, nil
 }
 
-// Delete deletes a specific tuple from the database table specified.  If no matching tuples are found an error will be
-// returned therefore the caller is responsible for checking for existing records.
-func Delete[T db.Model](ctx context.Context, db *pgxpool.Pool, uuid uuid.UUID) (int64, error) {
+// Exists checks whether a record exists in the table with a primary key that matches uuid.
+func Exists[T db.Model](ctx context.Context, db *pgxpool.Pool, uuid uuid.UUID) (bool, error) {
 	var record T
-	query := psql.Delete(
-		dm.From(record.TableName()),
-		dm.Where(psql.Quote(record.PrimaryKey()).EQ(psql.Arg(uuid))))
+	query := fmt.Sprintf("SELECT EXISTS(SELECT 1 FROM %s WHERE %s=%s)",
+		psql.Quote(record.TableName()), psql.Quote(record.PrimaryKey()), psql.Arg(uuid))
+
+	slog.Debug("executing query", "query", query, "args", uuid.String())
 
-	sql, params, err := query.Build()
+	var result bool
+	err := db.QueryRow(ctx, query, uuid).Scan(&result)
 	if err != nil {
-		return 0, fmt.Errorf("failed to build delete query for '%s/%s': %w", record.TableName(), uuid, err)
+		return false, fmt.Errorf("failed to execute query: %w", err)
 	}
 
-	result, err := db.Exec(ctx, sql, params...)
+	return result, nil
+}
+
+// Create creates a record of the requested model type.  The stored record is returned on success; otherwise an error
+// is returned.
+func Create[T db.Model](ctx context.Context, db *pgxpool.Pool, record T) (*T, error) {
+	nonNilTags := GetAllDBTagsFromStruct(record, ExcludeNilValues)
```

Comment:
I find this very confusing. Besides, it is only used in Create. It is polluting everything else. Why don't we simply accept columns and values as arguments? or if you insist on keeping this, why don't you create another function to get the tags whose values are not nil instead of adding an argument to GetAllDBTagsFromStruct? It's much cleaner in my opinion.

Reply:
I'll rework this so that there are 2 public functions that don't have this flag, but internally they will re-use the same code rather than duplicate it since it is mostly the same.

Reply:
fixed